roboteam_ai/src/control/ballHandling/BallHandlePosControl.cpp

```diff
@@ -201,23 +210,33 @@ RobotCommand BallHandlePosControl::handleBall(const Vector2 &targetBallPos, Trav
     status = HANDLING_BALL;
     // check if we are doing something already
     if (dribbleBackwards->getBackwardsProgression() != DribbleBackwards::START) {
-        return dribbleBackwards->getRobotCommand(robot, targetBallPos, targetAngle);
+        return updatePID(xBallHandlePID, yBallHandlePID,
+                dribbleBackwards->getRobotCommand(robot, targetBallPos, targetAngle));
     }
 
     if (dribbleForwards->getForwardsProgression() != DribbleForwards::START) {
-        return dribbleForwards->getRobotCommand(robot, targetBallPos, targetAngle);
+        return updatePID(xBallHandlePID, yBallHandlePID,
+                dribbleForwards->getRobotCommand(robot, targetBallPos, targetAngle));
     }
 
     switch (travelStrategy) {
-    case FORWARDS: return dribbleForwards->getRobotCommand(robot, targetBallPos, targetAngle);
-    case BACKWARDS: return dribbleBackwards->getRobotCommand(robot, targetBallPos, targetAngle);
+    case FORWARDS:
+        return updatePID(xBallHandlePID, yBallHandlePID,
+                dribbleForwards->getRobotCommand(robot, targetBallPos, targetAngle));
+
+    case BACKWARDS:
+        return updatePID(xBallHandlePID, yBallHandlePID,
+                dribbleBackwards->getRobotCommand(robot, targetBallPos, targetAngle));
+
     default:
     case NO_PREFERENCE: {
         // choose based on distance from the ball to the target
         if (ballIsFarFromTarget) {
-            return dribbleForwards->getRobotCommand(robot, targetBallPos, targetAngle);
+            return updatePID(xBallHandlePID, yBallHandlePID,
```

Comment:
The usage of this updatePID function is really vague. You're updating a PID but also passing a robotcommand?

Reply:
I'll change name of the function